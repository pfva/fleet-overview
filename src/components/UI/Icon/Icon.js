import styles from './Icon.module.css';

const Icon = ({ type }) => {
  if (type === 'active') {
    return <span className={styles.dot} data-testid="Icon-active"></span>;
  }
  return null;
};

export default Icon;
