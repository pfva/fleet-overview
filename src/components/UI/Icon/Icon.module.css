.dot {
  position: relative;
  top: 6px;
  display: inline-block;
  height: var(--font-size-extra-small);
  width: var(--font-size-extra-small);
  margin-right: var(--font-size-small);
  border-radius: 50%;
  background-color: var(--color-active);
  box-shadow: 0 0 5px var(--color-active);
}
