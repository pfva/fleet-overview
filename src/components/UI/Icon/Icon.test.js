import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

describe('Icon', () => {
  it('renders without crashing', () => {
    const tree = renderer.create(<Icon />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the acive icon when passed as a type prop', () => {
    render(<Icon type="active" />);
    const icon = screen.getByTestId(/Icon-active/i);
    expect(icon).toBeInTheDocument();
  });
});
