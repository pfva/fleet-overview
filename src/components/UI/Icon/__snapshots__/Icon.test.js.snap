// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders without crashing 1`] = `null`;
